modules/util/net_test.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package util
+
+import (
+	"net"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIsIPPPrivate(t *testing.T) {
+	cases := []struct {
+		ip        string
+		isPrivate bool
+	}{
+		// case 0
+		{
+			ip:        "127.0.0.1",
+			isPrivate: true,
+		},
+		// case 1
+		{
+			ip:        "127.1.2.3",
+			isPrivate: true,
+		},
+		// case 2
+		{
+			ip:        "10.255.255.0",
+			isPrivate: true,
+		},
+		// case 3
+		{
+			ip:        "8.8.8.8",
+			isPrivate: false,
+		},
+		// case 4
+		{
+			ip:        "::1",
+			isPrivate: true,
+		},
+		// case 4
+		{
+			ip:        "2a12:7c40::f00d",
+			isPrivate: false,
+		},
+	}
+
+	for n, c := range cases {
+		i, _ := net.ParseIP(c.ip)
```

Comment:
I don't think that we already should include go1.18 code, even for tests.

Reply:
This is go1.17 code: https://cs.opensource.google/go/go/+/refs/tags/go1.17:src/net/ip.go;l=704